components/channel_invite_modal/channel_invite_modal.tsx

```diff
@@ -310,17 +318,30 @@ export default class ChannelInviteModal extends React.PureComponent<Props, State
 
         const buttonSubmitText = localizeMessage('multiselect.add', 'Add');
         const buttonSubmitLoadingText = localizeMessage('multiselect.adding', 'Adding...');
+        let excludedAndNotInTeamUsers = this.props.profilesNotInCurrentTeam;
+        if (this.props.excludeUsers) {
+            excludedAndNotInTeamUsers = excludedAndNotInTeamUsers.concat(Object.values(this.props.excludeUsers));
+        }
 
-        let users = filterProfilesStartingWithTerm(this.props.profilesNotInCurrentChannel.concat(this.props.profilesInCurrentChannel), this.state.term).filter((user) => {
-            return user.delete_at === 0 &&
-                !this.props.profilesNotInCurrentTeam.includes(user as UserProfileValue) &&
-                (this.props.excludeUsers !== undefined && !this.props.excludeUsers[user.id]);
-        }).map((user) => user as UserProfileValue);
-
+        let users = this.filterOutDeletedAndExcludedAndNotInTeamUsers(
+            filterProfilesStartingWithTerm(
+                this.props.profilesNotInCurrentChannel.concat(this.props.profilesInCurrentChannel),
+                this.state.term),
+            excludedAndNotInTeamUsers);
         if (this.props.includeUsers) {
             const includeUsers = Object.values(this.props.includeUsers);
             users = [...users, ...includeUsers];
         }
+        users = [
+            ...this.filterOutDeletedAndExcludedAndNotInTeamUsers(
+                filterProfilesStartingWithTerm(this.props.profilesFromRecentDMs, this.state.term),
+                excludedAndNotInTeamUsers).
+                slice(0, USERS_FROM_DMS).
+                map((user) => user as UserProfileValue),
+            ...users,
+        ].
+            slice(0, MAX_USERS).
+            filter(((profile, index, profiles) => profiles.indexOf(profile) === index));
```

Comment:
Instead of filtering out the duplicates this way which loops over the whole array once for every `filter` call, we can actually use a Set to dedupe this array like in the selector. You can just replace this line with
```js
users = Array.from(new Set(users));
```
and it'll remove any duplicates and just keep the first time they appear